numpy/core/arrayprint.py

```diff
@@ -78,13 +78,20 @@ def _make_options_dict(precision=None, threshold=None, edgeitems=None,
     if legacy not in [None, False, '1.13']:
         warnings.warn("legacy printing option can currently only be '1.13' or "
                       "`False`", stacklevel=3)
+
     if threshold is not None:
         # forbid the bad threshold arg suggested by stack overflow, gh-12351
         if not isinstance(threshold, numbers.Number):
             raise TypeError("threshold must be numeric")
         if np.isnan(threshold):
             raise ValueError("threshold must be non-NAN, try "
                              "sys.maxsize for untruncated representation")
+
+    if precision is not None:
+        # forbid the bad precision arg as suggested by issue #18254
+        if not isinstance(precision, int):
```

Comment:
Perhaps better as `precision=operator.index(precision)`, else this blows up if users pass an `np.int32` or something else that's harmless